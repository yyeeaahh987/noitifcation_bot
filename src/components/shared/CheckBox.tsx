import React from "react";
import { ICONS } from "../../constants";

type Props = {
  checked: boolean;
  onChange: (a: boolean) => void;
  style?: any;
  isNetwork?: boolean;
};

const CheckBox = (props: Props) => {
  const { checked, onChange, style, isNetwork } = props;
  return (
    <div
      onClick={() => {
        onChange(!checked);
      }}
      style={{ cursor: "pointer", ...style }}
    >
      <img
        src={
          checked
            ? isNetwork
              ? ICONS.NETWORK_CHECKBOX_CHECKED
              : ICONS.CHECKBOX_CHECKED
            : isNetwork
            ? ICONS.NETWORK_CHECKBOX_EMPTY
            : ICONS.CHECKBOX_EMPTY
        }
        alt={checked ? "filled checkbox icon" : "empty checkbox icon"}
        style={{ display: "block" }}
      />
    </div>
  );
};

export default CheckBox;
