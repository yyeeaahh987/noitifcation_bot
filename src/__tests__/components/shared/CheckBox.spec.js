import { useState } from "react";
import renderer from "react-test-renderer";
import { screen, fireEvent, render } from "@testing-library/react";
import CheckBox from "../../../components/shared/CheckBox";

const CheckBoxTest = () => {
  const [checked, setChecked] = useState(false);

  const handleChange = (value) => {
    setChecked(value);
  };

  return <CheckBox checked={checked} onChange={handleChange} />;
};

describe("CheckBox component", () => {
  test("changes the state when clicked", () => {
    const component = renderer.create(<CheckBoxTest />);
    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();

    renderer.act(() => {
      tree.props.onClick();
    });

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("icon alt attribute is changed when clicked", () => {
    render(<CheckBoxTest />);

    expect(screen.getByAltText(/empty checkbox icon/i)).toBeTruthy();

    fireEvent.click(screen.getByAltText(/empty checkbox icon/i));

    expect(screen.getByAltText(/filled checkbox icon/i)).toBeTruthy();
  });
});
