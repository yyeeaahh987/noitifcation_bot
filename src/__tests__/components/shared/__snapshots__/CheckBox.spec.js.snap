// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox component changes the state when clicked 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <img
    alt="empty checkbox icon"
    src="/images/icons/checkbox-empty.svg"
    style={
      Object {
        "display": "block",
      }
    }
  />
</div>
`;

exports[`CheckBox component changes the state when clicked 2`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <img
    alt="filled checkbox icon"
    src="/images/icons/checkbox-checked.svg"
    style={
      Object {
        "display": "block",
      }
    }
  />
</div>
`;
